Source/WebCore/rendering/RenderCounter.cpp

```diff
@@ -165,23 +165,33 @@ static CounterDirectives listItemCounterDirectives(RenderElement& renderer)
 {
     if (is<RenderListItem>(renderer)) {
         auto& item = downcast<RenderListItem>(renderer);
-        if (auto explicitValue = item.explicitValue())
-            return { *explicitValue, std::nullopt };
-        return { std::nullopt, item.isInReversedOrderedList() ? -1 : 1 };
+        return {
+            .resetValue = std::nullopt,
+            .incrementValue = item.isInReversedOrderedList() ? -1 : 1,
+            .setValue = std::nullopt
+        };
     }
     if (auto element = renderer.element()) {
         if (is<HTMLOListElement>(*element)) {
             auto& list = downcast<HTMLOListElement>(*element);
-            return { list.start(), list.isReversed() ? 1 : -1 };
+            return {
+                .resetValue = list.start(),
+                .incrementValue = list.isReversed() ? 1 : -1,
```

Comment:
I'm confused by this returning of 1 if the list is reversed (which is in the existing code too). Why do we do this?

Reply:
Probably a typo, I'll try to add a testcase for this.

Reply:
To my surprise, this is actually correct, there's some strange ordering in LayoutTests/fast/css/counters/counter-list-item.html otherwise.